pkg/lib/serverTime.js

```diff
@@ -321,13 +316,166 @@
+        const text = `# This file is automatically generated by Cockpit\n\n[Time]\n${config.enabled ? "" : "#"}NTP=${config.servers.join(" ")}\n`;
 
         return cockpit.spawn(["mkdir", "-p", "/etc/systemd/timesyncd.conf.d"], { superuser: "try" })
                 .then(() => custom_ntp_config_file.replace(text));
+    }
+
+    function get_ntp_config_chronyd() {
+        const chrony_file = cockpit.file("/etc/chrony.conf", { superuser: "try" });
+
+        return chrony_file.read()
+                .then(function(text) {
+                    const result = {
+                        backend: "chronyd",
+                        supported: true,
+                        standard: [],
+                        sources: [],
+                        extra: []
+                    };
+
+                    let in_cockpit_section = false;
+                    for (const line of text.split("\n")) {
+                        if (line == "# Added by Cockpit")
+                            in_cockpit_section = true;
+                        else {
+                            const fields = line.split(" ");
+                            if (fields[0] == "server") {
+                                if (in_cockpit_section)
+                                    result.extra.push(fields[1]);
+                                else
+                                    result.standard.push({ name: fields[1], enabled: true });
+                            } else {
+                                in_cockpit_section = false;
+                                if (fields[0] == "pool")
+                                    result.standard.push({ name: fields[1], enabled: true });
+                                if (fields[0] == "#pool" || fields[0] == "#server")
+                                    result.standard.push({ name: fields[1], enabled: false });
+                                if (fields[0] == "sourcedir")
+                                    result.sources.push({ name: fields[1], enabled: true });
+                                if (fields[0] == "#sourcedir")
+                                    result.sources.push({ name: fields[1], enabled: false });
+                            }
+                        }
+                    }
+
+                    return result;
+                });
+    }
+
+    function set_ntp_config_chronyd(config) {
+        const chrony_file = cockpit.file("/etc/chrony.conf", { superuser: "try" });
+
+        function confline(line, list, name) {
+            const c = list.find(e => e.name == name);
+            if (!c)
+                return line;
+            const line_trim = line.trim();
+            const line_enabled = !line_trim.startsWith("#");
+            if (c.enabled && !line_enabled)
+                return line_trim.substr(1);
+            if (!c.enabled && line_enabled)
+                return "#" + line_trim;
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17992-20230113-125341-c8c3ecb6-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)